### PR TITLE
Test Moderated Sessions RBAC engine for multiple session types per test case

### PR DESCRIPTION
**Pull request by xacrimon**
This PR adds logic to test both SSH and K8S sessions for many of the test cases. Note that we don't do this for all test cases as there are semantic differences between SSH and K8S RBAC logic and some are specifically designed to catch this behaviour.

### CODE REVIEW

File: lib/auth/session_access_test.go [hunk 1]
```diff
@@ -149,9 +149,50 @@ func failFilterStartTestCase(t *testing.T) startTestCase {
...
+				Username: "participant2",
+				Roles:    []types.Role{participantRole},
+				Mode:     "peer",
+			},
+		},
+		expected: false,
+	}
+}
+
+func failVersionCheckStartTestCase(t *testing.T) startTestCase {
```
Comment (marcoandredinis) [inline lib/auth/session_access_test.go:171]: The unit tests are failing because this method is unused

Comment (xacrimon) [reply lib/auth/session_access_test.go:171]: Thanks, that got lost in my cherry-pick, curiously VSCode didn't complain about this. Fixed.

File: lib/auth/session_access_test.go [hunk 2]
```diff
@@ -164,7 +164,26 @@ func failFilterStartTestCase(t *testing.T) startTestCase {
 				Mode:     "peer",
 			},
 		},
-		expected: false,
+		expected: []bool{false},
+	}
+}
+
+func versionFallbackStartTestCase(t *testing.T) startTestCase {
+	hostRole, err := types.NewRoleV3("host", types.RoleSpecV5{})
```
Comment (marcoandredinis) [inline lib/auth/session_access_test.go:172]: Why are we using `NewRoleV3` here and in the `versionDefaultJoinTestCase`?
I would expect a call to `NewRole`

**Review by marcoandredinis — APPROVED** [review]
LGTM
Just left a question regarding the usage of `NewRoleV3`

File: lib/auth/session_access_test.go (hunk 2 above)
Comment (xacrimon) [reply lib/auth/session_access_test.go:172]: Ah, the reason we're calling this here, is that this testcase makes sure that a participant with no v5 roles retains the same privileges as pre-moderated-sessions join RBAC tightening.

**Review by smallinsky — APPROVED** [review]

**Pull request merged**